pymc3/distributions/distribution.py

```diff
@@ -98,7 +98,7 @@ def __new__(cls, name, *args, **kwargs):
             raise TypeError("observed needs to be data but got: {}".format(type(data)))
         total_size = kwargs.pop("total_size", None)
 
-        dims = kwargs.pop("dims", None)
+        dims = kwargs["dims"] if "dims" in kwargs else None
```

Comment:
This is a breaking change to the `pm.Distribution` API. Right now you can define new distributions that do not accept a dims kwargs, but with this they have to accept it. Maybe we can live with this, but at the very least this should be mentioned in the release notes somewhere.

Reply:
I reverted that change, as we're now using `__new__` to enable the `zerosum_dims` kwarg